keras/preprocessing/image.py

```diff
@@ -164,6 +164,42 @@ def random_channel_shift(x, intensity, channel_axis=0):
     return x
 
 
+def rgb_to_bgr(x, data_format=None):
+    if data_format is None:
+        data_format = K.image_data_format()
+    assert data_format in {'channels_last', 'channels_first'}
+
+    if data_format == 'channels_first':
+        # 'RGB'->'BGR'
+        x = x[:, ::-1, :, :]
+    else:
+        # 'RGB'->'BGR'
+        x = x[:, :, :, ::-1]
+    return x
+
+
+def subtract_mean_pixel(x, mean_pixel, data_format=None):
+    try:
+        if len(mean_pixel) == 3:
```

Comment:
The idea was to use a try...catch to know if mean_pixel is a list with 3 numbers or only a number so the grayscale support is included. Also you can use a number as mean_pixel with 3 channel images an it will apply that mean_pixel to all three channels.